mm2src/coins/utxo/utxo_common.rs

```diff
@@ -1641,6 +1746,48 @@ pub fn check_all_inputs_signed_by_pub(tx: &UtxoTx, expected_pub: &[u8]) -> Resul
     Ok(true)
 }
 
+pub fn watcher_validate_taker_fee<T: UtxoCommonOps>(
+    coin: T,
+    taker_fee_hash: Vec<u8>,
+    verified_pub: Vec<u8>,
+) -> ValidatePaymentFut<()> {
+    let fut = async move {
+        let mut attempts = 0;
+        let taker_fee_hash = H256Json::from(taker_fee_hash.as_slice());
+        loop {
+            let taker_fee_tx = match coin
```

Comment:
Please add more validations that are done in `utxo_common::validate_fee` on the next iteration. As of now, a trading party can submit any transaction signed by its pubkey, and it will pass the validation. It might be not even a dex fee transaction.

Reply:
Alright you can merge the current state. I was working on making the watchers periodically check if the taker already spent the maker payment, thought I could make it in this sprint but that also requires some review so it's better I leave it to the next sprint, together with the final notes here I didn't work on.

Reply:
I think we should also check for the transaction time or block height since a trading party can reuse an old fee transaction signed by his pub and it will also pass validation without this check.